### PR TITLE
CLIENTS: Add PATCH methods

### LINKED ISSUE #134: CLIENTS: Add PATCH methods
Some APIs use PATCH instead of PUT and RESTFulSense should support that. Currently, neither client in RESTFulSense exposes PATCH methods.

### PR DESCRIPTION
**Pull request by diogod3**
Closes #134

### CODE REVIEW

**Comment by cjdutoit** [comment]
LGTM

File: RESTFulSense/Clients/IRESTFulApiClient.cs [hunk 1]
```diff
@@ -89,6 +89,27 @@ public interface IRESTFulApiClient
 
         ValueTask<T> PutContentAsync<T>(string relativeUrl);
         ValueTask<T> PutContentAsync<T>(string relativeUrl, CancellationToken cancellationToken);
+        ValueTask<T> PatchContentAsync<T>(
+            string relativeUrl,
+            T content,
+            string mediaType = "text/json",
+            bool ignoreDefaultValues = false);
```
Comment (cjdutoit) [inline RESTFulSense/Clients/IRESTFulApiClient.cs:96]: Missing the following that was added to the IRESTFulApiFactoryClient

```
        ValueTask<T> PatchContentAsync<T>(
            string relativeUrl,
            T content,
            CancellationToken cancellationToken,
            string mediaType = "text/json",
            bool ignoreDefaultValues = false);
```

File: RESTFulSense/Clients/RESTFulApiClient.cs [hunk 2]
```diff
@@ -263,6 +262,92 @@ public async ValueTask<T> PutContentAsync<T>(string relativeUrl, CancellationTok
...
+            HttpContent contentString = ConvertToHttpContent(content, mediaType, ignoreDefaultValues);
+
+            HttpResponseMessage responseMessage =
+               await PatchAsync(relativeUrl, contentString);
+
+            await ValidationService.ValidateHttpResponseAsync(responseMessage);
+
+            return await DeserializeResponseContent<T>(responseMessage);
+        }
+
```
Comment (cjdutoit) [inline RESTFulSense/Clients/RESTFulApiClient.cs:280]: Missing
```
        public async ValueTask<T> PatchContentAsync<T>(
            string relativeUrl,
            T content,
            CancellationToken cancellationToken,
            string mediaType = "text/json",
            bool ignoreDefaultValues = false)
        {
            HttpContent contentString = ConvertToHttpContent(content, mediaType, ignoreDefaultValues);

            HttpResponseMessage responseMessage =
               await this.httpClient.PatchAsync(relativeUrl, contentString, cancellationToken);

            await ValidationService.ValidateHttpResponseAsync(responseMessage);

            return await DeserializeResponseContent<T>(responseMessage);
        }
```

**Review by cjdutoit — CHANGES_REQUESTED** [review]
IRESTFulApiFactoryClient and IRESTFulApiClient not in sync

**Comment by diogod3** [comment]
Thanks @cjdutoit

I used the PUT methods as a guideline and that method was missing from the interface, but it was present in the implementation 😅 I added it for both PUT and PATCH.

**Comment by SlimAhmad** [comment]
@cjdutoit  would this be merged anytime soon, i'll like to make use of the patch method in my library

**Pull request closed without merging**